hrm-domain/packages/hrm-search-config/generateElasticsearchQuery.ts

```diff
@@ -193,15 +197,17 @@ const generateQueriesFromId = <TDoc extends DocumentType>({
   const queries = terms
     .map(term => {
       // Ignore terms that are not entirely a number, as that breaks term queries against integer fields
-      if (Number.isNaN(Number(term))) {
+      if (Number.isNaN(Number(term)) || !Number.isInteger(term)) {
```

Comment:
I don't think this will catch the specific issue we were seeing where the phone number was interpreted as a number but then rejected for being _too large_

```
Number.isInteger(11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111)
-> true
Number.isNaN(11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111)
-> false
```

I think we also need an explict range check?

Reply:
@stephenhand Good catch.
From [Postgres docs](https://www.postgresql.org/docs/current/datatype-numeric.html):
> integer	4 bytes	typical choice for integer	-2147483648 to +2147483647
And from [ElasticSearch docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html)
> integer A signed 32-bit integer with a minimum value of -2^(31) and a maximum value of 2^(31)-1.

Since 2^(31) = 2147483648, I think it will be safe to assume that this check would solve this particular issue:
`Number.parseFloat(term, 10) > 2147483648`

For example
```
Number.parseFloat('11111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111', 10) > 2147483648
-> true
```

Is this aligned with your thoughts?